Automattic/wp-calypso
# Checkout: Ensure correct thank-you page is shown to plan customers who don't transfer a domain

#### Changes proposed in this Pull Request

* This PR reworks the checkout logic such that all cases except domain transfers see the standard check-out screen.

**Before**

<img width="1680" alt="Screen Shot 2019-11-08 at 12 03 54 PM" src="https://user-images.githubusercontent.com/2124984/68511410-e833e180-0243-11ea-9358-f8ee9f3f3b96.png">

**After**

<img width="1680" alt="Screen Shot 2019-11-08 at 4 45 41 PM" src="https://user-images.githubusercontent.com/2124984/68512742-3d252700-0247-11ea-8216-db193befa8df.png">


#### Testing instructions

* Switch to this PR
* Create a brand-new WordPress.com account with a Free plan
* Go to Manage -> Plugins and click the upsell to Business
* Add a Business plan to the cart and check out
* You should see the "After" screenshot as a thank-you, not the "Before"

Fixes #37009
